### ISSUE TITLE
Consider adding a Replayable interface for projectors

### ISSUE THREAD

**Issue by sebastiandedeyne** [spatie/laravel-event-projector#58]
When replaying a projector that doesn't have a `resetState` method, a method not found exception is thrown that's hard to debug for first-timers.

Requiring a `Replayable` interface would make this more explicit. Alternatively, do a `method_exists` call and provide an error message to clarify that the package user must implement `resetState`.

**Comment by freekmurze** [comment]
Can do!

**Comment by freekmurze** [comment]
Seems like we already have a dedicated `CouldNotResetProjector` exception that gets thrown when a projector doesn't have a `resetState` method. If has [a very clear message]().  We also already have [a test](https://github.com/spatie/laravel-event-projector/blob/320f9fb0fa0a4a01a6de89f9b99e1d8f0197523b/tests/ProjectorTest.php#L47-L53) for this.

In what situation a method not found exception is thrown?

**Comment by sebastiandedeyne** [comment]
Hmm it was that method that I saw. My problem with the message is that I wasn't sure whether it was something I had to implement _myself_ or something in the package or configuration. Maybe reword it so some sort of "Did you implement a `resetState` method?
